File: backend/app/presentation/schemas/photos.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class PhotoItemResponse(BaseModel):
    id: UUID
    preview_url: str
    original_url: str
    created_at: datetime
    size: int = Field(ge=0)


class PhotoListResponse(BaseModel):
    items: list[PhotoItemResponse]
    total: int = Field(ge=0)
    has_next: bool

File: frontend/src/types/photos.ts
export interface PhotoItem {
  id: string;
  preview_url: string;
  original_url: string;
  created_at: string;
  size: number;
}

export interface PhotoListResponse {
  items: PhotoItem[];
  total: number;
  has_next: boolean;
}

export const PHOTO_BATCH_SIZE = 30;
